tasks/run_agent.rb

```diff
@@ -0,0 +1,160 @@
+#!/opt/puppetlabs/puppet/bin/ruby
+
+require 'open3'
+require 'etc'
+require 'json'
+require 'yaml'
+require_relative '../../ruby_task_helper/files/task_helper.rb'
+
+class RunAgentTask < TaskHelper
+  def task(puppet_bin: '/opt/puppetlabs/bin/puppet', wait_time: 300, _noop: false, **_kwargs)
+    @puppet_bin = puppet_bin
+    @wait_time = wait_time
+    @noop = _noop
+
+    check_running_as_root
+    check_agent_not_disabled
+    wait_for_puppet_lockfile
+    run_puppet
+  end
+
+  def wait_for_puppet_lockfile
+    waited = 0
+    while agent_locked? && waited < @wait_time
+      sleep 1
+      waited += 1
+    end
+
+    if agent_locked?
+      raise TaskHelper::Error.new(
+        'Lockfile still exists after waiting',
+        'theforeman-puppet/lockfile_timeout_expired'
+      )
+    end
+
+    waited
+  end
+
+  def check_agent_not_disabled
+    if agent_disabled?
+      raise TaskHelper::Error.new(
+        'Agent is disabled on this node',
+        'theforeman-puppet/agent_disabled',
+        disabled_reason
+      )
+    end
+  end
+
+  def agent_locked?
+    File.exist?(agent_catalog_run_lockfile)
+  end
+
+  def agent_disabled?
+    File.exist?(agent_disabled_lockfile)
+  end
+
+  def disabled_reason
+    file = File.open agent_disabled_lockfile
```

Comment:
Isn't this equivalent to `JSON.load(File.read(agent_disabled_lockfile))`?

Reply:
yes. Updated